system-addon/bootstrap.js

```diff
@@ -156,4 +157,11 @@ this.shutdown = function shutdown(data, reason) {
   modulesToUnload.forEach(Cu.unload);
 };
 
-this.uninstall = function uninstall(data, reason) {};
+this.uninstall = function uninstall(data, reason) {
+  // Uninstall is also called before an upgrade, so we
+  // need to make sure this is an actual uninstall
+  if (activityStream && reason === REASON_ADDON_UNINSTALL) {
+    activityStream.uninstall(reason);
```

Comment:
Any particular reason to skip upgrades (and downgrades)? The list of prefs can be reduced when installing a different version, although the impact isn't too bad as it'll just appear as if there's a default pref when it's most likely not being used.